simplified triangular peak

jspaaks opened the pull request:
In this PR I simplified the triangular peak function:
- `location` now refers to location of the peak
- no more leading or trailing widths
- no more asserts
- argument `width` removed
- `width_base_left` and `width_base_right` are now required arguments
- `location` is a required argument
- removed `sign`

Also added new tests

Tests are based on `linspace` (more suited to the expected input type of float)

Refs #47, https://github.com/sequgen/sequgen/issues/61#issuecomment-811050775

sverhoeven reviewed (changes requested):
I like the changes, much clearer how to use.

What is the unit of the width?
Based on the value of the time axis or the index positions of the time axis.

sverhoeven commented on tests/deterministic/test_triangular_peak.py:
```diff
@@ -1,34 +1,27 @@
+    expected = np.array([0, 0.5, 1.0, 0.75, 0.5, 0.25, 0., 0., 0., 0., 0.])
+    default_height = 1.0
+    assert actual[t_predict == location] == default_height, "expected maximum height at t == location"
+    assert_almost_equal(actual, expected)
```
What happens when you use 0 for width_base_*. For example I want a signal that starts suddenly and fades linearly overtime.

```suggestion
    assert_almost_equal(actual, expected)

def test_with_zero_base_left():
    t_predict = np.linspace(0, 5, 11)  # array([0. , 0.5, 1. , 1.5, 2. , 2.5, 3. , 3.5, 4. , 4.5, 5. ])
    actual = triangular_peak(t_predict, width_base_left=0.0, width_base_right=2.0, location=1.0)
    expected = np.array([])
    assert_almost_equal(actual, expected)

def test_with_zero_base_right():
    t_predict = np.linspace(0, 5, 11)
    actual = triangular_peak(t_predict, width_base_left=2.0, width_base_right=0.0, location=3.0)
    expected = np.array([])
    assert_almost_equal(actual, expected)
```

The actual values resp are:
```
E        x: array([0.  , 0.  , 1.  , 0.75, 0.5 , 0.25, 0.  , 0.  , 0.  , 0.  , 0.  ])
E        x: array([0.  , 0.  , 0.  , 0.25, 0.5 , 0.75, 0.  , 0.  , 0.  , 0.  , 0.  ])
```

The actual value in test_with_zero_base_left makes sense to me.
However for test_with_zero_base_right I hoped to see 
```
E        x: array([0.  , 0.  , 0.25, 0.5 , 0.75, 1.0, 0.  , 0.  , 0.  , 0., 0.  ])
``

Can you add those tests and fill the expected values?

jspaaks commented:
> What is the unit of the width?

Always in units of t_predict, updated docstring

> 0 for width_base_*

Not supported, but you can put in a small value like 1e-9 instead of 0 (what constitutes 'small' depends on the use case). Added an `assert`.


I'll give your tests a whirl, hold on

jspaaks commented:
Thanks for the review
